webharness/harness.js

```diff
@@ -60,6 +63,62 @@ function socket_message(data) {
 }
 
 /*
+ * Receive a single message from |socket|. If
+ * there is a deferred (from wait_for_message)
+ * waiting on it, resolve that deferred. Otherwise
+ * queue the message for a future waiter.
+ */
+function socket_message_test(data) {
+  var message = JSON.parse(data);
+  if (socket_message_promises_test.length > 0) {
+    var res = socket_message_promises_test.shift();
+    res(message);
+  } else {
+    socket_messages_test.push(message);
+  }
+}
+
+function connect_socket_test() {
```

Comment:
Looks like connect_socket_test() and connect_socket() have again the same functionality. If possible they should share common code.